File: browser/src/model/wage.ts
export interface WageItem {
  id: number; //
  days: number; // 带薪天数
  truename: string; // 姓名
  idCard: string; // 身份证号码
  base: number; // 基本工资
  bonus: number; // 奖金
  preTax: number; // 税前总金额
  postTax: number; // 税后总金额
  bonusDetail: string; // 奖金详情
  year: number; // 时间
  month: number; // 时间
}
export interface BonusTypeItem {
  id:number; //
  name:string; // 奖金名称
  status:string; // 状态
  gap:number;//间隔月份
  createBy:number; // 创建者
  createTime:Date; // 创建时间
  updateBy:number; // 更改者
  updateTimeL:Date; // 更改时间
}

export interface BonusItem {
  id:number; //
  uid:number; // 用户
  btid:number; // 奖金类型

  sum:number; // 金额
  year:number; // 发放年份
  month:number; // 发放月份

  createBy:number; // 创建者
  createTime:Date; // 创建时间
  updateBy:number; // 更改者
  updateTimeL:Date; // 更改时间
}

File: Server/src/main/java/cn/edu/zqu/hr_system/project/model/entities/Wage.java
package cn.edu.zqu.hr_system.project.model.entities;

import com.alibaba.excel.annotation.ExcelIgnore;
import com.alibaba.excel.annotation.ExcelProperty;

import cn.edu.zqu.hr_system.project.base.BaseEntity;
import lombok.Data;
@Data
public class Wage extends BaseEntity {
  @ExcelProperty(value = "姓名")
  String truename; // 姓名
  @ExcelProperty(value = "身份证号码")
  String idCard; // 身份证号码
  @ExcelProperty(value = "基本工资")
  Double base; // 基本工资
  @ExcelProperty(value = "奖金")
  Double bonus; // 奖金
  @ExcelProperty(value = "税前工资")
  Double preTax; // 税前总金额
  @ExcelProperty(value = "税后工资")
  Double postTax; // 税后总金额
  @ExcelProperty(value = "奖金详情")
  String bonusDetail; // 奖金详情
  @ExcelProperty(value = "工资天数")
  Integer days; // 工资天数
  @ExcelIgnore
  Integer year; // 发放年份
  @ExcelIgnore
  Integer month; // 发放月份
}

